### PR TITLE
make the check for `mpirun` optional in OpenMPI easyblock

### PR DESCRIPTION
**Pull request by damianam**

### CODE REVIEW

File: easybuild/easyblocks/o/openmpi.py [hunk 1]
```diff
@@ -160,11 +160,12 @@ def load_module(self, *args, **kwargs):
     def sanity_check_step(self):
         """Custom sanity check for OpenMPI."""
 
-        bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'mpirun', 'ompi_info', 'opal_wrapper']
+        bin_names = ['mpicc', 'mpicxx', 'mpif90', 'mpifort', 'ompi_info', 'opal_wrapper']
         if LooseVersion(self.version) >= LooseVersion('5.0.0'):
             bin_names.append('prterun')
```
Comment (boegel) [inline easybuild/easyblocks/o/openmpi.py:165]: @damianam We're now no longer checking for `mpirun` for OpenMPI v5.0.0, is that intentional?

File: easybuild/easyblocks/o/openmpi.py [hunk 2]
```diff
@@ -160,11 +160,12 @@ def load_module(self, *args, **kwargs):
...
         else:
-            bin_names.append('orterun')
+            if '--with-orte=no' not in self.cfg['configopts'] and '--without-orte' not in self.cfg['configopts']:
```
Comment (boegel) [inline easybuild/easyblocks/o/openmpi.py:167]: Can you clarify this a bit? Does configuring without `orte` imply not having `mpirun` at all?

Comment (damianam) [reply easybuild/easyblocks/o/openmpi.py:167]: Exactly. `mpirun` is not built. `orterun` is built, but at least IMO not really used and probably should not be built either.

Comment (akesandgren) [reply easybuild/easyblocks/o/openmpi.py:167]: without orterun for older versions you can't start openmpi code. It's always used.

**Comment by damianam** [comment]
This was relevant a bit ago, but according to https://github.com/spack/spack/issues/44902#issuecomment-2194977313 it is no longer desirable for 5.x, so I think we can drop this.

**Pull request closed without merging**